dask/dot.py

```diff
@@ -131,37 +132,82 @@ def to_graphviz(dsk, data_attributes=None, function_attributes=None):
     return g
 
 
-def dot_graph(dsk, filename='mydask', **kwargs):
-    g = to_graphviz(dsk, **kwargs)
+def _get_display_cls(format):
+    """
+    Get the appropriate IPython display class for `format`.
 
-    if filename is not None:
-        g.save(filename + '.dot')
+    Returns `IPython.display.SVG` if format=='svg', otherwise
+    `IPython.display.Image`.
 
-        try:
-            check_call('dot -Tpdf {0}.dot -o {0}.pdf'.format(filename),
-                       shell=True)
-            check_call('dot -Tpng {0}.dot -o {0}.png'.format(filename),
-                       shell=True)
+    If IPython is not importable, return dummy function that swallows its
+    arguments and returns None.
+    """
+    dummy = lambda *args, **kwargs: None
+    try:
+        import IPython.display as display
+    except ImportError:
+        # Can't return a display object if no IPython.
+        return dummy
+
+    if format in {'pdf', 'dot'}:
```

Comment:
This is a py26 compatibility violation.  I suspect that using a tuple here would be just as good.